### PR TITLE
refactor(PVS/V1048): remove duplicated assignments

### PR DESCRIPTION
**Pull request by merrittlj**
Yet another PR removing unnecessary assignments. In this one, I refactored multiple files. Should be the last PR like this for awhile.

### CODE REVIEW

File: src/nvim/eval/funcs.c [hunk 1]
```diff
@@ -7343,7 +7343,7 @@ static void f_settagstack(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
   // third argument: action - 'a' for append and 'r' for replace.
   // default is to replace the stack.
   if (argvars[2].v_type == VAR_UNKNOWN) {
-    action = 'r';
+    // action = 'r';
```
Comment (dundargoc) [inline src/nvim/eval/funcs.c:7346]: You should remove the line instead of just commenting it out.

File: src/nvim/insexpand.c [hunk 2]
```diff
@@ -2890,7 +2890,7 @@ static int process_next_cpt_value(ins_compl_next_state_T *st, int *compl_type_ar
     status = INS_COMPL_CPT_END;
   } else {
     if (ctrl_x_mode_line_or_eval()) {
-      compl_type = -1;
+      // compl_type = -1;
```
Comment (dundargoc) [inline src/nvim/insexpand.c:2893]: Same here.

**Review by dundargoc — CHANGES_REQUESTED** [review]

File: src/nvim/eval/funcs.c (hunk 1 above)
Comment (zeertzjq) [reply src/nvim/eval/funcs.c:7346]: Why?

Comment (dundargoc) [reply src/nvim/eval/funcs.c:7346]: Lol, I guess not

**Pull request merged**